### PR TITLE
feat: forbid string interpolation

### PR DESCRIPTION
**Pull request by cjdcordeiro**
- [x] Have you signed the [CLA](http://www.ubuntu.com/legal/contributors/)?

-----

With the recent introduction of `environment`, users might expect string interpolation to work such that the following would be valid:

```yaml
environment:
  FOO: BAR
  BAZ: $FOO
```

but **it is not**. The YAML loading takes all map values as literal strings and we do not yet have any string interpolation for variable substitution.

We will though, but while that feature is being designed and implemented, it is safer to limit the user assumptions and forbid the use of bash-like variables in the `environment` key values.

### CODE REVIEW

**Review by lengau — APPROVED** [review]

File: tests/unit/test_project.py [hunk 1]
```diff
@@ -197,6 +197,26 @@ def test_project_unmarshal_with_unsupported_fields(unsupported_field, yaml_loade
     )
 
 
+@pytest.mark.parametrize(
+    "variable,is_forbidden",
+    [("$BAR", True), ("BAR_$BAZ", True), ("BAR$", False)],
```
Comment (SamirPS) [inline tests/unit/test_project.py:202]: How ``BAR$`` is not forbidden ?

Comment (cjdcordeiro) [reply tests/unit/test_project.py:202]: why should it be? if it is the last char, then it is just a regular char.

Comment (SamirPS) [reply tests/unit/test_project.py:202]: Yeah true,  sorry

Comment (cjdcordeiro) [reply tests/unit/test_project.py:202]: ;)

**Review by mr-cal — APPROVED** [review]

**Review by sergiusens — APPROVED** [review]

**Pull request merged**